server/src/main/java/org/opensearch/search/profile/query/CollectorResult.java

```diff
@@ -107,6 +155,11 @@ public CollectorResult(StreamInput in) throws IOException {
         this.collectorName = in.readString();
         this.reason = in.readString();
         this.time = in.readLong();
+        this.reduceTime = in.readLong();
```

Comment:
@ticheng-aws couple of points here:
 - the new fields have to be read (and added) to the end of the stream
 - please use version guards to deal with mixed clusters

```
        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
              this.reduceTime = in.readLong();
              this.maxSliceTime = in.readLong();
              this.minSliceTime = in.readLong();
              this.avgSliceTime = in.readLong();
              this.sliceCount = in.readVInt();
        } else {
              this.maxSliceTime = 0;
              this.minSliceTime = 0;
              this.avgSliceTime = 0;
              this.sliceCount = 0;
        }
```

Reply:
Hi @reta, I appreciate your review. Could you kindly assist me by reviewing it again?